### ISSUE TITLE
Issue with server-side validation of repeater fields

### ISSUE THREAD

**Issue by michaelhue** [verbb/formie#730]
**Description**

I received multiple bug reports from people trying to submit our Formie form. During my investigation I noticed a problem with the server-side validation of fields inside a repeater when trying to submit a form more than once.

We have a repeater for entering multiple social media profile links, containing two required fields:
1. Social Media Site (Dropdown with options like "Facebook", "Instragram", etc.)
2. Social Media URL (Text input)

When the user adds a repeater row, fills in both fields correctly and then triggers a validation error elsewhere, the repeater fields will always be empty and invalid after subsequent submission attempts.

This is the `$_POST` data for the first submission with a validation error (the required field  "name1" is left empty, resulting in a validation error):

```php
$_POST = [
    'action' => 'formie/submissions/submit'
    'handle' => 'foerderatlas'
    'siteId' => '1'
    'fields' => [
        'name1' => ''
        'socialMediaProfile' => [
            'rows' => [
                'new1' => [
                    'fields' => [
                        'socialMediaSeite' => 'Facebook'
                        'socialMediaUrl' => 'https://example.org'
                    ]
                ]
            ]
            'sortOrder' => [
                0 => 'new1'
            ]
        ]
]
```

After filling out the "name1" field and submitting again:

```php
$_POST = [
    'action' => 'formie/submissions/submit'
    'handle' => 'foerderatlas'
    'siteId' => '1'
    'fields' => [
        'name1' => 'John Doe'
        'socialMediaProfile' => [
            'rows' => [
                0 => [
                    'fields' => [
                        'socialMediaSeite' => 'Facebook'
                    ]
                ]
                1 => [
                    'fields' => [
                        'socialMediaUrl' => 'https://example.org'
                    ]
                ]
            ]
            'sortOrder' => [
                0 => ''
            ]
        ]
]
```

Notice how the rows become disconnected. There is now another validation for the fields "socialMediaSeite" and "socialMediaUrl". Both fields are now empty on the front end, every subsequent submission will fail.

Sorry if the explanation is unclear, let me know if there are any questions.

**Steps to reproduce**
1. Create a form with a repeater field, containing one dropdown field and one text field.
2. Disable client-side validation.
3. Submit the form with at least one repeater row, but make sure a validation error is triggered by omitting a required field.
4. Submit the form again.
5. Notice how the repeater fields don't contain the submitted values anymore and are now always invalid on subsequent submissions.

### Form settings
Please pick from the options below.

- Multi-page form: No
- Submission Method: Page Reload
- Client-side Validation: No
- Custom Form Templates: No

**Additional info**
- Plugin version: 1.5.7
- Craft version: Craft Pro 3.7.30.1
- Multi-site: No

**Comment by michaelhue** [comment]
I made a quick screencap showing the issue:

https://user-images.githubusercontent.com/31861/151141575-ec944c4d-60bd-4d81-8e14-a11b4f7d3c3d.mov

**Comment by engram-design** [comment]
Thanks for that, that all makes sense, and sorry for the issue. Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.7",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.5.8](https://github.com/verbb/formie/releases/tag/1.5.8)

**Comment by engram-design** [comment]
Fixed in [1.5.8](https://github.com/verbb/formie/releases/tag/1.5.8)
